### PR TITLE
Fix dependency management

### PR DESCRIPTION
**Pull request by cristianonicolai**
Delay calling service to reload dependencies until changes are saved.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@etirelli @manstis @Rikkola please have a look, seems to fix for me but feel free to amend if I'm missing something here.

**Comment by manstis** [comment]
I'll let @Rikkola review, but (a) the change appears to circumnavigate quite a lot of other Presenter logic (and I'm not sure of the impact), (b) what is the issue this PR resolves?

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/dependencies/DependencyGrid.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public Widget asWidget() {
 
     public void onAddDependency() {
         pom.getDependencies().add( new Dependency() );
-        show();
+        view.show( pom.getDependencies() );
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/dependencies/DependencyGrid.java]: This is correct.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/dependencies/DependencyGrid.java [hunk 2]
```diff
@@ -92,7 +92,7 @@ public void onAddDependencyFromRepository() {
 
     public void onRemoveDependency( final Dependency dependency ) {
         pom.getDependencies().remove( dependency );
-        show();
+        view.show( pom.getDependencies() );
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/dependencies/DependencyGrid.java]: Not sure about this one. Need to play around with this.

Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/dependencies/DependencyGrid.java]: This is correct. No need to update the dependencies from the servers side if we are just removing one from the list.

**Comment by Rikkola** [comment]
+1

**Comment by psiroky** [comment]
Merged into 6.4.x. Please make sure to submit PR for master as well.

**Pull request closed without merging**